std/experimental/allocator/package.d

```diff
@@ -943,6 +960,24 @@ pure nothrow @nogc
     int[] a = [1, 2, 4];
     uninitializedFillDefault(a);
     assert(a == [0, 0, 0]);
+
+    char[] b = [1, 2, 4];
+    uninitializedFillDefault(b);
+    assert(b == [0xff, 0xff, 0xff]);
+
+    wchar[] c = [1, 2, 4];
+    uninitializedFillDefault(c);
+    assert(c == [0xffff, 0xffff, 0xffff]);
+}
+
+@system unittest
+{
+    static struct P { float x = 0; float y = 0; }
+
+    static assert(isInitAllZeroBits!P);
+    P[] a = [P(10, 11), P(20, 21), P(40, 41)];
+    uninitializedFillDefault(a);
+    assert(a == [P.init, P.init, P.init]);
```

Comment:
Needs tests for `const`, e.g

```d
static struct S { float x = 0; float y = 0; }
static foreach (e; ["mutable", "non-mutable"])
{{

    static if (e == "mutable")
        alias P = S;
    else
        alias P = ConstOf!S;

    static assert(isInitAllZeroBits!P);
    P[] a = [P(10, 11), P(20, 21), P(40, 41)];
    uninitializedFillDefault(a);
    assert(a == [P.init, P.init, P.init]);
}}
```

fails with:

```
std/experimental/allocator/package.d(929): Error: function core.stdc.string.memset(return scope void* s, int c, ulong n) is not callable using argument types (const(S)*, int, ulong)
std/experimental/allocator/package.d(994): Error: template instance std.experimental.allocator.uninitializedFillDefault!(const(S)) error instantiating
```

Reply:
Shouldn't trying to write to `const` without casting const-ness away result in compilation failure?